src/main/java/org/broadinstitute/hellbender/utils/dataflow/DataflowUtils.java

```diff
@@ -67,6 +73,50 @@ public void processElement(ProcessContext c) {
                 .apply(ParDo.of(new LoadReadsFromFileFn(intervals)));
     }
 
+    /**
+     * Ingest BAM files from a Hadoop file system and load them into a
+     * <code>PCollection<Read></code>.
+     * @param pipeline a configured Pipeline
+     * @param intervals intervals to select reads from
+     * @param path Hadoop file or directory path to read BAM files from
+     * @return a <code>PCollection<Read></code> with all the reads that overlap the
+     * given intervals in the BAM files
+     */
+    public static PCollection<Read> getReadsFromHadoopBams(final Pipeline pipeline, final List<SimpleInterval> intervals, final String path) {
+        PCollection<KV<LongWritable, SAMRecordWritable>> input =
+            (PCollection<KV<LongWritable, SAMRecordWritable>>) pipeline.apply(
+                HadoopIO.Read.from(path).withFormatClass(AnySAMInputFormat.class)
+                    .withKeyClass(LongWritable.class).withValueClass(SAMRecordWritable.class));
+        return input.apply(ParDo.of(new DoFn<KV<LongWritable, SAMRecordWritable>, Read>() {
+            @Override
+            public void processElement(ProcessContext c) throws Exception {
+                SAMRecord record = c.element().getValue().get();
+                if (overlaps(record, intervals)) {
+                    c.output(ReadConverter.makeRead(record));
+                }
+            }
+        }));
+    }
+
+    /**
+     * Tests if a given record overlaps any interval in a collection.
+     */
+    private static boolean overlaps(SAMRecord record, List<SimpleInterval> intervals) {
```

Comment:
This is an unfortunate omission from the SimpleInterval API.   It's come up in a number of places that it's needed.  

The interval lists can be very large (100k+ intervals), so iterating over each interval is a bit scary in terms of performance.  We should adapt the IntervalTree from htsjdk so that we can do this operation efficiently, but that's outside of the scope of this pull request.